cmd/mdatagen/templates/metrics.go.tmpl

```diff
@@ -60,6 +60,51 @@ func DefaultMetricsSettings() MetricsSettings {
 	}
 }
 
+// ResourceAttributeSettings provides common settings for a particular metric.
+type ResourceAttributeSettings struct {
+	Enabled bool `mapstructure:"enabled"`
+
+	enabledProvidedByUser bool
+}
+
+// IsEnabledProvidedByUser returns true if `enabled` option is explicitly set in user settings to any value.
+func (ras *ResourceAttributeSettings) IsEnabledProvidedByUser() bool {
+	return ras.enabledProvidedByUser
+}
+
+func (ras *ResourceAttributeSettings) IsEnabled() bool {
+	return ras.enabledProvidedByUser || ras.Enabled
+}
+
+func (ras *ResourceAttributeSettings) Unmarshal(parser *confmap.Conf) error {
+	if parser == nil {
+		return nil
+	}
+	err := parser.Unmarshal(ras, confmap.WithErrorUnused())
+	if err != nil {
+		return err
+	}
+	ras.enabledProvidedByUser = parser.IsSet("enabled")
+	return nil
+}
+
+// ResourceAttributesSettings provides settings for {{ .Name }} resource attributes.
+type ResourceAttributesSettings struct {
+	{{- range $name, $attr := .ResourceAttributes }}
+	{{ $name.Render }} ResourceAttributeSettings `mapstructure:"{{ $name }}"`
+	{{- end }}
+}
```

Comment:
I think we should wrap `MetricsSettings` and `ResourceAttributesSettings` into another structure like  `MetricsBuilderConfig` and add it to the component configs with `mapstructure:",squash"`. And it should be passed to the `NewMetricsBuilder` instead of `MetricsSettings` and `WithResourceAttributesSettings`.